### PR TITLE
refactor(entities-*): migrate vault picker to entities-vault [KM-718]

### PR DESCRIPTION
**Pull request by 2eha0**
# Summary

preview: https://pr-4926--gateway-manager.cloud-preview.konghq.tech/gateway-manager

KM-718

### CODE REVIEW

File: packages/entities/entities-vaults/src/components/VaultSecretPickerProvider.vue [hunk 1]
```diff
@@ -1,8 +1,5 @@
 <template>
-  <div
-    v-if="props.schema.referenceable"
```
Comment (2eha0) [inline packages/entities/entities-vaults/src/components/VaultSecretPickerProvider.vue:3]: Not necessary, it already has `v-if` outside

File: packages/entities/entities-vaults/src/components/VaultSecretPickerProvider.vue [hunk 2]
```diff
@@ -20,13 +17,17 @@
...
 
 <script setup lang="tsx">
-import type { AutofillSlotProps } from '@kong-ui-public/forms'
 import composables from '../composables'
 
-const props = defineProps<AutofillSlotProps>()
+type Props = {
+  value: string
+  update: (value: string) => void
+}
```
Comment (2eha0) [inline packages/entities/entities-vaults/src/components/VaultSecretPickerProvider.vue:25]: Making the props simpler, it doesn't need `props.schema` anymore

File: packages/entities/entities-vaults/src/components/VaultSecretPicker.vue [hunk 3]
```diff
@@ -133,9 +133,9 @@ const { i18n: { t } } = composables.useI18n()
 
 const props = defineProps({
   config: {
-    type: Object as PropType<KonnectPluginFormConfig | KongManagerPluginFormConfig>,
+    type: Object as PropType<KonnectBaseFormConfig | KongManagerBaseFormConfig>,
     required: true,
-    validator: (config: KonnectPluginFormConfig | KongManagerPluginFormConfig): boolean => {
+    validator: (config: KonnectBaseFormConfig | KongManagerBaseFormConfig): boolean => {
```
Comment (2eha0) [inline packages/entities/entities-vaults/src/components/VaultSecretPicker.vue:138]: Using the more generic types `KonnectBaseFormConfig` and `KongManagerBaseFormConfig` instead

**Review by TT1228 — APPROVED** [review]
LGTM

**Pull request merged**